src/mscorlib/src/System/String.cs

```diff
@@ -2577,6 +2581,150 @@ public static int Compare(String strA, String strB, StringComparison comparisonT
             }                        
         }
 
+#if FEATURE_CORECLR
+        [Pure]
+        private unsafe Boolean StartsWithOrdinal(String startsWith)
+        {
```

Comment:
Take a look at String EqualsHelper. It has yet the loop for the same intent, but in different style. It would be really useful to use uniform style everywhere.